src/x/headers/headers.go

```diff
@@ -0,0 +1,123 @@
+	// metrics being written or read.
+	// In the form of a storage policy string, e.g. "1m:14d".
+	// Only required if the metrics type header does not specify unaggregated
+	// metrics type.
+	MetricsStoragePolicyHeader = M3HeaderPrefix + "Storage-Policy"
+
+	// RestrictByTagsJSONHeader provides tag options to enforces on queries,
+	// in JSON format. See `handler.stringTagOptions` for definitions.`
+	RestrictByTagsJSONHeader = M3HeaderPrefix + "Restrict-By-Tags-JSON"
+
+	// MapTagsByJSONHeader provides the ability to mutate tags of timeseries in
+	// incoming write requests. See `MapTagsOptions` for structure.
+	MapTagsByJSONHeader = M3HeaderPrefix + "Map-Tags-JSON"
+
+	// LimitMaxSeriesHeader is the M3 limit timeseries header that limits
+	// the number of time series returned by each storage node.
+	LimitMaxSeriesHeader = M3HeaderPrefix + "Limit-Max-Series"
+
+	// LimitMaxDocsHeader is the M3 limit docs header that limits
+	// the number of docs returned by each storage node.
+	LimitMaxDocsHeader = M3HeaderPrefix + "Limit-Max-Docs"
+
+	// LimitRequireExhaustiveHeader is the M3 limit exhaustive header that will
+	// ensure M3 returns an error if the results set is not exhaustive.
+	LimitRequireExhaustiveHeader = M3HeaderPrefix + "Limit-Require-Exhaustive"
+
+	// UnaggregatedStoragePolicy specifies the unaggregated storage policy.
+	UnaggregatedStoragePolicy = "unaggregated"
+
+	// DefaultServiceEnvironment is the default service ID environment.
+	DefaultServiceEnvironment = "default_env"
+	// DefaultServiceZone is the default service ID zone.
+	DefaultServiceZone = "embedded"
+
+	// HeaderClusterEnvironmentName is the header used to specify the
+	// environment name.
+	HeaderClusterEnvironmentName = "Cluster-Environment-Name"
+	// HeaderClusterZoneName is the header used to specify the zone name.
+	HeaderClusterZoneName = "Cluster-Zone-Name"
+	// HeaderDryRun is the header used to specify whether this should be a dry
+	// run.
+	HeaderDryRun = "Dry-Run"
+	// HeaderForce is the header used to specify whether this should be a forced
+	// operation.
+	HeaderForce = "Force"
+
+	// LimitHeader is the header added when returned series are limited.
+	LimitHeader = M3HeaderPrefix + "Results-Limited"
+
+	// LimitHeaderSeriesLimitApplied is the header applied when fetch results
+	// are maxed.
+	LimitHeaderSeriesLimitApplied = "max_fetch_series_limit_applied"
+
+	// RenderFormat is used to switch result format for query results rendering.
+	RenderFormat = M3HeaderPrefix + "Render-Format"
+
+	// DisableJSONDisallowUnknownFields is header if set to true that allows
+	// for clients to send fields unknown by a HTTP/JSON endpoint and still
+	// parse the request, this is helpful for sending a request with a new
+	// schema to an older instance and still have it respond successfully
+	// using the fields it knows about.
+	DisableJSONDisallowUnknownFields = "Disable-JSON-Disallow-Unknown-Fields"
```

Comment:
Should we add the `M3HeaderPrefix` to this? I know we're a bit inconsistent in how we apply it above but I think that's because we had already used the old form.

Reply:
Ah yes, that was an accident.